include/swift/AST/Evaluator.h

```diff
@@ -416,6 +463,51 @@ class Evaluator {
     return result;
   }
 
+private:
+  template <typename Request,
+            typename std::enable_if<!Request::isDependencySink>::type * = nullptr>
+  void reportEvaluatedResult(const Request &r,
+                             const typename Request::OutputType &o) const {}
+
+  template <typename Request,
+            typename std::enable_if<Request::isDependencySink>::type * = nullptr>
+  void reportEvaluatedResult(const Request &r,
+                             const typename Request::OutputType &o) {
+    r.writeDependencySink(*this, o);
+  }
```

Comment:
I have just read both of these over about four times, and I finally spotted the "const" difference. Sigh... C++. But now I'm wondering why it's OK to not write the sink if the receiver happens to be `const`. Would it be possible to put in a comment explaining this? Is there a possibility of creating a bug by calling the first one when a write is really needed because of a bogus "const" on the receiver of the message?